src/scope.rs

```diff
@@ -209,6 +209,12 @@ where
 
             self.data = Some(data);
         }
+
+        if !cfg.extensions.is_empty() {
+            let mut data = self.data.unwrap_or_else(Extensions::new);
+            data.extend(cfg.extensions);
+            self.data = Some(data);
+        }
```

Comment:
im not sure this is more efficient than simply: `self.data.extend(cfg.extensions)`

Reply:
What if `self.data` is `None`?

Reply:
an if let with a mut ref then?

Reply:
you mean something like this?

```rust
        if let Some(ref mut data) = self.data {
            data.extend(cfg.extensions);
        } else {
            self.data = Some(cfg.extensions);
        }
```

Reply:
If you want to be really concise:

```
self.data.get_or_insert_with(Extensions::new).extend(cfg.extensions);
```